projects/plutus/src/fault_monitor.c

```diff
@@ -51,3 +52,24 @@ StatusCode fault_monitor_init(FaultMonitorStorage *storage, const FaultMonitorSe
 
   return ltc_afe_request_cell_conversion(storage->settings.ltc_afe);
 }
+
+bool fault_monitor_process_event(FaultMonitorStorage *storage, const Event *e) {
+  switch (e->id) {
+    case PLUTUS_EVENT_AFE_FAULT:
+      LOG_DEBUG("AFE FSM fault %d\n", e->data);
+      bps_heartbeat_raise_fault(storage->settings.bps_heartbeat,
+                                EE_BPS_HEARTBEAT_FAULT_SOURCE_LTC_AFE_FSM);
+
+      // Attempt to recover from failure
+      ltc_afe_request_cell_conversion(storage->settings.ltc_afe);
+      break;
+    case PLUTUS_EVENT_AFE_CALLBACK_RUN:
+      bps_heartbeat_clear_fault(storage->settings.bps_heartbeat,
+                                EE_BPS_HEARTBEAT_FAULT_SOURCE_LTC_AFE_FSM);
+      break;
+    default:
+      return false;
+  }
+
+  return true;
```

Comment:
Nit: Just have the cases `return true` it is more consistent if the cases either all `return` or all `break`.